js-core/homeworks/homework-8/main.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+
+/*
+ * TASK 1----------------------------------------------
+ * Создайте функцию которая будет запоминать переданные
+ * ей аргументы, и прибавлять их в строку
+ * и отображать в консоле всю строку
+ * */
+
+function solution1() {
+	let str = '';
+	return function(value){
+		return str += `${value} `;
+	}
+}
+
+let stringBuffer = solution1();
+
+console.log('task 1-1--->', stringBuffer('Замыкания')); // Замыкания
+console.log('task 1-2--->', stringBuffer('Использовать нужно')); // Замыкания Использовать нужно
+console.log('task 1-3--->', stringBuffer('Привет')); // Замыкания Использовать нужно Привет вызываем много раз
+
+/*
+ * TASK 2--------------------------------------------------
+ * Напишите функцию которая принимает в качестве аргумента строку
+ * из скобочек и посчитайте, что все скобочки закрываются корректно
+ * */
+//
+//если в строке нечетное количество символов, вернуть false
+//
+//
+//
+function validBraces(str) { /// тут не смогла вовремя выйти из рекурсии))
+	let etalon = ['()','{}','[]'];
```

Comment:
pretty complex solution :)

could you please think, can we use .replace method here and maybe it would help somehow to you?